third_party/pcsc-lite/naclport/server_clients_management/src/clients_manager.cc

```diff
@@ -146,15 +169,35 @@ bool PcscLiteServerClientsManager::DeleteHandlerMessageListener ::
   return true;
 }
 
+PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    UpdateAdminPolicyMessageListener(AdminPolicyGetter* admin_policy_getter)
+    : admin_policy_getter_(admin_policy_getter) {}
+
+std::string PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    GetListenedMessageType() const {
+  return kUpdateAdminPolicyMessageType;
+}
+
+bool PcscLiteServerClientsManager::UpdateAdminPolicyMessageListener::
+    OnTypedMessageReceived(Value data) {
+  const UpdateAdminPolicyMessageData message_data =
+      ConvertFromValueOrDie<UpdateAdminPolicyMessageData>(std::move(data));
```

Comment:
Another interesting aspect here is that when one adds a new policy in the future, they might forget about registering it in `StructValueDescriptor<AdminPolicy>::GetDescription()`, which will lead to breaking old policies too.

I'll say it's not a blocker for your PR, just something that came to my mind and that we might want to fix eventually. (I suppose it can be something like adding something like `StructValueDescriptor::PermitUnknownKeys()`.)